fs/proc/pid.c

```diff
@@ -111,7 +132,7 @@ static int proc_pid_cmdline_show(struct proc_entry *entry, struct proc_data *buf
         goto out_free_task;
 
     size_t size = task->mm->argv_end - task->mm->argv_start;
-    uint8_t *data = malloc(size);
+    char *data = malloc(size);
```

Comment:
What's wrong with uint8_t?

Reply:
You know, if `CHAR_BIT != 8`

Reply:
Is that really allowed?

C99 section 5.2.4.2.1:
> Their implementation-defined values shall be equal or greater in magnitude (absolute value) to those shown, with the same sign.
> `CHAR_BIT` = 8: number of bits for smallest object that is not a bit-field (byte)

section 7.18.1.1:
> The typedef name `uint`*N*`_t` designates an unsigned integer type with width *N*, no padding bits, and a two’s complement representation. Thus, `uint24_t` denotes a signed integer type with a width of exactly 24 bits.
> These types are optional.

So `CHAR_BIT` must be at least 8, and can only be greater than 8 if no 8-bit non-bitfield type exists. `uint8_t` is optional, but if it does exist, then `CHAR_BIT` is required to be 8.

Reply:
Yeah but something something aliasing